### ISSUE TITLE
The frontmatter template produces headers whose validity depends on the parser: `skill: [plugin:name]` loads under PyYAML and fails under libyaml

### ISSUE THREAD

**Issue by juniordurant** [rebelytics/one-skill-to-rule-them-all#146]
**Target:** `SKILL.md` — "File format" ("Every prose value is double-quoted"),
and the step-2 suspect check.
**Verified at:** upstream `main` (`7518a85`) — `SKILL.md`,
`references/observation-log.md`, `references/environments.md` and
`references/skill-authoring.md` at HEAD are byte-identical to the v3.2.0 tag,
so this is not a stale install.
**Adjacent, not duplicate:** #81 (the scan certifies frontmatter it never
parses) is the parent — this is the class it left behind. PR #134 extends the
suspect check and explicitly *withdraws* the flow-collection check, listing
"`[a: b]` inside a flow list" under known limits; that decision rests on a
PyYAML-only ground truth, which is exactly the measurement that cannot see this.
#79 (migrate-log.py rejected scoped `plugin:skill` identifiers) is the same
identifier shape in the converter rather than in the template.

## What happened

Upgrading a 200-entry log to v3.2.0 and running the new step-2 check: **8
suspect headers**. A `YAML.safe_load` pass over the same files immediately
afterwards: **127 failing**. The obvious conclusion — the heuristic under-reports
by 15x — turned out to be the wrong one.

Measured on one corpus of 202 files, the same day, three ways:

| instrument | verdict |
|---|---|
| the shipped awk check | 8 suspect |
| PyYAML 6.0.3 | **0 invalid** |
| Ruby Psych 3.1.0 (libyaml) | **127 invalid** |

All three numbers are correct. The 127 are almost all this line, which is the
commonest line in the log:

```yaml
skill: [superpowers:verification-before-completion, code-review]
```

YAML 1.2 allows `:` inside a plain scalar in flow context when the next
character is not a space, so PyYAML reads it as the string it looks like.
libyaml applies the 1.1 rule that `:` ends a plain scalar in flow context
regardless, and raises `found unexpected ':' while scanning a plain scalar`.
Both are correct implementations of different specifications, so the header is
neither valid nor invalid on its own — it is **not portable**.

Reproduction, no log required:

```bash
python3 -c 'import yaml; print(yaml.safe_load("skill: [a:b]"))'   # {'skill': ['a:b']}
ruby    -ryaml -e 'p YAML.safe_load("skill: [a:b]")'              # Psych::SyntaxError
python3 -c 'import yaml; print(yaml.safe_load("skill: [\"a:b\"]"))'  # both accept
ruby    -ryaml -e 'p YAML.safe_load("skill: [\"a:b\"]")'             # the portable form
```

## Why the current text did not prevent it

The template produces it. "Quote the value … **keep lists in `[]` with bare
kebab-case names**" is correct advice that silently stops being correct for a
plugin-scoped name, because a scoped name is not bare kebab-case — it contains a
colon. Every scoped entry written by following the template is in the
non-portable form, in the `skill:` field that the review's work-queue pass, the
session-start scan and any hook read first.

The failure is invisible from either side. A Python consumer (`migrate-log.py`,
`validate-skill-bundle.py`, most hooks) sees a clean log forever. A Ruby, Go or
`yq` consumer sees two thirds of the log fail to load. Nothing in the skill says
which dialect the format targets, so neither side has a reason to look.

## Suggested fix

1. **Make the template portable**, and show a scoped name in it, since the
   template is what gets copied:

   > Quote any list entry containing a colon — `skill: ["superpowers:test-driven-development", task-observer]` —
   > or use a block list, where the colon is legal unquoted. A plugin-scoped
   > name is not a bare kebab-case name; `[plugin:name]` loads under PyYAML and
   > raises under libyaml-backed parsers (Ruby Psych, Go, `yq`).

   Both forms parse identically under both dialects, so this costs two
   characters per scoped entry and removes the class.

2. **Name the dialect** the frontmatter targets, once, wherever the format is
   defined. "Valid" is a claim about a parser; a format used from more than one
   language needs the intersection, not whichever parser the author had.

3. **In the fallback check**, add the flow-list entry (a `[`-value whose
   items contain a colon once quoted substrings are removed) and label the
   number a floor. PR #134 withdrew the flow check for good reasons — flow
   *maps* have many corners and `[a: b]` really is valid — but the narrow
   "entry containing a colon" case is decidable and is the one that actually
   occurs, because the template generates it. A minimal form that agrees with
   libyaml on 202 files here, with 0 false positives on the 7-case control set
   (quoted scoped entry, block list, bare block value, plain list, fully quoted
   prose with an inner colon, and the two genuinely broken shapes):

   ```awk
   fm && /^[a-z_]+: *\[/ {v=$0; sub(/^[a-z_]+: */,"",v); gsub(/"[^"]*"/,"",v); gsub(/\047[^\047]*\047/,"",v)
                          if (v ~ /:/) {print FILENAME; nextfile}}
   ```

4. **Report the instrument with the number.** The line that made this hard to
   see was `suspect: 8` standing alone. "8 suspect", "0 invalid" and "127
   invalid" describe the same corpus.

The generalisable half, if it is worth a line in `skill-authoring.md`'s
pre-flight: *a second instrument that CONTRADICTS a strong prior gets
investigated; one that CONFIRMS it never does.* The 127 confirmed what the
existing text already implies about heuristics, which is precisely why it went
unchallenged for an hour — the truthful reading was not "the check misses 119
errors" but "one parser rejects what another accepts, and the check models
neither".

---

Related (checked before filing, not duplicates): #81, #134, #79, #107.

---

Reported from a production install of v3.2.0 (20 project copies, a 200-entry
log). Prepared with Claude Code (Claude Opus 5); every command quoted above was
executed verbatim from a clean `sh` before posting, and a human reviewed and
approved this report.

**Comment by rebelytics** [comment]
Thanks — four solid reports from you in this batch. This one's queued for one of the next weekly releases.
